src/content.js

```diff
@@ -1,28 +1,29 @@
-import injectGithub from './injections/github';
 
-function init() {
-  const observer = new MutationObserver((mutations) => {
-    for (let i = 0; i < mutations.length; i++) {
-      const mutation = mutations[i];
+import { observe } from 'selector-observer';
+import fileIcons from 'file-icons-js';
+import { DEFAULT_ICON } from './constants';
 
-      if (mutation.type === 'childList') {
-        const target = mutation.target;
+chrome.storage.sync.get('isColor', ({ isColor }) => {
+  function injectGithub(item) {
+    const isFile = item.querySelector('.octicon-file');
+    const name = item.querySelector('.js-navigation-open').textContent;
+    const icon = document.createElement('span');
 
-        if ($(target).has(`.Box-row`).length) {
-          injectGithub(target);
-        }
+    if (isFile) {
+      let className = fileIcons.getClass(name) || DEFAULT_ICON;
+
+      if (isColor) {
+        className = fileIcons.getClassWithColor(name) || DEFAULT_ICON;
       }
-    }
-  });
-  observer.observe(document.body, {
-    attributes: true,
-    childList: true,
-    characterData: true,
-    subtree: true
-  });
 
-  injectGithub('body');
-}
+      icon.classList.add('octicon-file', 'gfi', className);
```

Comment:
When the `isColor` is set to true, the `className` could contain multiple classes separated by space, i.e. "github-icon medium-red" which causes the following error:

```
 Failed to execute 'add' on 'DOMTokenList': The token provided ('git-icon medium-red') contains HTML space characters, which are not valid in tokens
```

Therefore, it needs to be split by space and spread to the `classist.add` function

Reply:
I didn't know how to switch color so I didn't test it. Thanks for the reminder!